project/frontend/src/components/PreferenceForm.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import getRecommendation from '../scripts/recommendation_script.js';
+
+class PreferenceForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      cuisine: [],
+      radius: {
+        pref: null,
+        weight: 3,
+      },
+      diningExp: {
+        pref: null,
+        weight: 3,
+      },
+      priceLevel: {
+        pref: null,
+        weight: 3,
+      },
+      currLocation: {
+        lat: null,
+        lng: null,
+      },
+      open: true,
+    };
+    this.changeState = this.changeState.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  // Updates the state of the input element so it holds the chosen value.
+  changeState(event) {
+    const field = this.state[event.target.name];
+    if (event.target.name === 'cuisine') {
+      const cuisineList = this.state.cuisine;
+      cuisineList.push(event.target.value);
+      this.setState({ [event.target.name]: cuisineList });
+    } else if (event.target.name === 'open') {
+      this.setState({ [event.target.name]: event.target.checked });
+    } else if (event.target.name === 'currLocation') {
+      field[event.target.id] = parseFloat(event.target.value);
+      this.setState({ [event.target.name]: field });
+    } else if (event.target.name === 'diningExp') {
+      field['pref'] = event.target.value;
+      this.setState({ [event.target.name]: field });
+    } else {
+      field['pref'] = parseInt(event.target.value);
+      this.setState({ [event.target.name]: field });
+    }
+  }
+
+  handleSubmit(event) {
+    getRecommendation(/* preferences= */ this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    const cuisines = ['Italian ', 'Mexican ', 'Indian '];
+    const distances_in_miles = {
+      '1 mile': 1,
+      '5 miles': 5,
+      '10 miles': 10,
+      '25 miles': 25,
+    };
+    const dining_experiences = {
+      Takeout: 'meal_takeaway',
+      Delivery: 'meal_delivery',
+      'Eat In': 'restaurant',
```

Comment:
is it intentional that there are quotes around only 'Eat in' ?

Reply:
I believe it's because of the space (I didn't write this part but this is what I assumed)

Reply:
Would it be better to change them all to having quotes so it's uniform?